src/Lecture2.hs

```diff
@@ -97,7 +112,9 @@ spaces.
 
 🕯 HINT: look into Data.Char and Prelude modules for functions you may use.
 -}
-dropSpaces = error "TODO"
+dropSpaces:: [Char] -> [Char]
+dropSpaces = reverse . dropWhile isSpace . reverse . dropWhile isSpace
```

Comment:
`reverse` is a rather slow function and it hangs on infinite lists so it's better to avoid it 🐌 

Have a look at `takeWhile`